editor/src/components/canvas/canvas-strategies/flex-resize-width-height-strategy.tsx

```diff
@@ -0,0 +1,146 @@
+import { MetadataUtils } from '../../../core/model/element-metadata-utils'
+import { ElementInstanceMetadataMap, JSXElement } from '../../../core/shared/element-template'
+import { CanvasVector, offsetPoint } from '../../../core/shared/math-utils'
+import { ElementPath } from '../../../core/shared/project-file-types'
+import { withUnderlyingTarget } from '../../editor/store/editor-state'
+import { EdgePosition } from '../canvas-types'
+import { setSnappingGuidelines } from '../commands/set-snapping-guidelines-command'
+import { updateHighlightedViews } from '../commands/update-highlighted-views-command'
+import { GuidelineWithSnappingVector } from '../guideline'
+import { CanvasStrategy } from './canvas-strategy-types'
+
+import { getMultiselectBounds } from './shared-absolute-move-strategy-helpers'
+import {
+  resizeBoundingBox,
+  runLegacyAbsoluteResizeSnapping,
+  createResizeCommands,
+} from './shared-absolute-resize-strategy-helpers'
+import { AbsoluteResizeControl } from '../controls/select-mode/absolute-resize-control'
+
+export const flexResizeWidthStrategy: CanvasStrategy = {
+  id: 'FLEX_RESIZE_WIDTH_HEIGHT',
+  name: 'Flex Resize (Width/Height)',
+  isApplicable: (canvasState, interactionState, metadata) => {
+    if (
+      canvasState.selectedElements.length === 1 &&
+      !interactionState?.interactionData.modifiers.alt &&
+      !interactionState?.interactionData.modifiers.shift // shift is aspect ratio locked resize implemented in absolute-resize-bounding-box-strategy.tsx
+    ) {
+      return canvasState.selectedElements.every((element) => {
+        const elementMetadata = MetadataUtils.findElementByElementPath(metadata, element)
+
+        return elementMetadata?.specialSizeMeasurements.position !== 'absolute'
```

Comment:
I think here you should also check if the parent has flex layout, the `specialSizeMeasurements` contains data for this

Reply:
@enidemi so, the parent would need to be `display: flex;`?

Reply:
yes! it's called parentLayoutSystem